File: Python/Clothing.py
# Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
# kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
# telah dispesifikasikan. Aamiin.

# ----- Latihan Python ----- #

# Import File
from Product import Product

# Membuat kelas Clothing dengan pewarisan dari Product
class Clothing(Product):
    # Membuat beberapa atribut private
    # size Clothing
    __size = ""
    # material Clothing
    __material = ""
    # gender Clothing
    __gender = ""

    # Constructor with default initialization
    def __init__(self, id="", name="", brand="", price="", size="", material="", gender=""):
        # memanggil konstruktor dari kelas induk(superclass)
        super().__init__(id, name, brand, price)
        self.__size = size
        self.__material = material
        self.__gender = gender

    # Setter and Getter methods for private attributes
    # Setter Size
    def setSize(self, size):
        self.__size = size

    # Getter Size
    def getSize(self):
        return self.__size

    # Setter Material
    def setMaterial(self, material):
        self.__material = material

    # Getter Material
    def getMaterial(self):
        return self.__material
    
    # Setter Gender
    def setGender(self, gender):
        self.__gender = gender

    # Getter Gender
    def getGender(self):
        return self.__gender

File: PHP/Clothing.php
<!-- 
    Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
    kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
    telah dispesifikasikan. Aamiin. 
-->

<!-- ----- Latihan PHP ----- -->

<?php
// import file
require('Product.php');

// Membuat kelas Clothing dengan pewarisan dari Product
class Clothing extends Product
{
    // Membuat atribut Private dari kelas Clothing
    private $size = '';
    private $material = '';
    private $gender = '';


    # Constructor with default initialization
    public function __construct($id = '', $name = '', $brand = '', $price = '', $image, $size = '', $material = '', $gender = '')
    {
        # memanggil konstruktor dari kelas induk(superclass)
        parent::__construct($id, $name, $brand, $price, $image);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    # Setter and Getter methods for private attributes

    # Set Size
    public function setSize($size)
    {
        $this->size = $size;
    }

    # Get Size
    public function getSize()
    {
        return $this->size;
    }

    # Set Material
    public function setMaterial($material)
    {
        $this->material = $material;
    }

    # Get Material
    public function getMaterial()
    {
        return $this->material;
    }

    # Set Gender
    public function setGender($gender)
    {
        $this->gender = $gender;
    }

    # Get Gender
    public function getGender()
    {
        return $this->gender;
    }
}